server/datastore/mysql/migrations/tables/20230721161508_QueriesDataMigrator_test.go

```diff
@@ -0,0 +1,294 @@
+			(6,'2023-07-21 20:34:34','2023-07-21 20:34:34',2,NULL,1800,1,NULL,NULL,NULL,NULL,'backup_tool_perf','backup_tool_perf','Track the percentage of total CPU time utilized by $backup_tool',NULL,''),
+
+			-- Global pack
+			(7,'2023-07-21 20:34:46','2023-07-21 20:34:46',1,2,86400,1,0,'','',NULL,'per_query_perf','per_query_perf','',NULL,''),
+			(8,'2023-07-21 20:34:51','2023-07-21 20:34:51',1,2,86400,1,0,'','',NULL,'per_query_perf','per_query_perf-1','',NULL,''),
+
+			-- Team-1 pack
+			(9,'2023-07-21 20:36:08','2023-07-21 20:36:08',3,6,86400,1,0,'','',NULL,'User 1 Query','User 1 Query','',NULL,''),
+			(10,'2023-07-21 20:36:13','2023-07-21 20:36:13',3,6,86400,1,0,'','',NULL,'User 1 Query','User 1 Query-1','',NULL,''),
+			(11,'2023-07-21 20:36:25','2023-07-21 20:36:25',3,2,86400,1,0,'','',NULL,'per_query_perf','per_query_perf','',NULL,''),
+
+			-- Team-2 pack
+			(12,'2023-07-21 20:36:50','2023-07-21 20:36:50',4,5,86400,1,0,'','',NULL,'backup_tool_perf','backup_tool_perf','',NULL,'');
+	`
+	_, err := db.Exec(dataStmts)
+	require.NoError(t, err)
+
+	// Apply current migration.
+	applyNext(t, db)
+
+	// 'User 2 Query' is non-scheduled and was created by user#3, so it should exists in both the
+	// global team and on team#2
+	stmt := "SELECT description, query, author_id, saved, observer_can_run, team_id, team_id_char FROM queries WHERE name = ?"
+	rows, err := db.Query(stmt, "User 2 Query")
+	require.NoError(t, err)
+	defer rows.Close()
+
+	var nRows int
+	var teamIDs []uint
+	var teamIDStrs []string
+
+	for rows.Next() {
+		nRows += 1
+		var teamIDStr string
+		query := fleet.Query{}
+		err := rows.Scan(
+			&query.Description,
+			&query.Query,
+			&query.AuthorID,
+			&query.Saved,
+			&query.ObserverCanRun,
+			&query.TeamID,
+			&teamIDStr,
+		)
+		require.NoError(t, err)
+		require.Equal(t, query.Description, "Some desc")
+		require.Equal(t, query.Query, "SELECT * FROM osquery_info;")
+		require.Equal(t, *query.AuthorID, uint(3))
+		require.Equal(t, query.Saved, true)
+		require.Equal(t, query.ObserverCanRun, true)
+
+		teamIDStrs = append(teamIDStrs, teamIDStr)
+		if query.TeamID != nil {
+			teamIDs = append(teamIDs, *query.TeamID)
+		}
+	}
+	require.Equal(t, nRows, 2)
+	require.ElementsMatch(t, teamIDStrs, []string{"", "2"})
+	require.Contains(t, teamIDs, uint(2))
+
+	// The global pack has 4 different schedules two targeting 'Admin Global Query' and the other
+	// two targeting 'performance-metrics' so I expect to see 6 queries here:
+	// 'Admin Global Query - 1' <- For schedule with id 1
+	// 'Admin Global Query - 2' <- For schedule with id 2
+	// 'performance-metrics' <- Original (kept because is referenced by an 2017 pack)
+	// 'performance-metrics - 7' <- For schedule with id 7
+	// 'performance-metrics - 8' <- For schedule with id 8
```

Comment:
Nit: name of the other query seems to be `per_query_perf`, not `performance-metrics`.

Reply:
Done in https://github.com/fleetdm/fleet/pull/12855/commits/e7c1518aa7d5c44cdf6195fc1e3eaf253839c0a9